go/merge/three_way.go

```diff
@@ -11,7 +11,19 @@ import (
 	"github.com/attic-labs/noms/go/types"
 )
 
-// ErrMergeConflict indicates that a merge attempt failed and must be resolved manually for the provided reason.
+// ResolveFunc is the type for custom merge-conflict resolution callbacks.
+// When the merge algorithm encounters two non-mergeable changes (aChange and
+// bChange) at the same path, it calls the ResolveFunc passed into ThreeWay().
+// The callback gets the types of the two incompatible changes (added, changed
+// or removed) and the two Values that could not be merged (if any). If the
```

Comment:
But it's actually the two Values that couldn't be merged; if the code is trying to merge two maps {foo: bar} and {foo: baz}, `a` will be "bar" and `b` wil be "baz". And the thing that's being returned is the new value to use for 'foo', in the above example.